telemetry/telemetry_record.py

```diff
@@ -96,25 +174,389 @@ def failed_url(url, output):
+
+        s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        try:
+            os.remove("/tmp/telemetrySocket")
+        except OSError:
+            pass
+        s.bind("/tmp/telemetrySocket")
+        s.listen(1)
+
+        p = Popen(cmd.format(str(i) + '_pc'), shell=True)
+
+        conn, addr = s.accept()
+        tmp_results = ''
+        while 1:
+            data = conn.recv(1024)
+            tmp_results += data
+            if not data:
+                break
+        conn.close()
+
+        p.wait()  # We can only load 1 url on Chrome at a time
+
+        # Parse data
+        benchmark_results = eval(tmp_results)
+
+        output = {urls[i]: {'cold_times': {trial_key: benchmark_results}}}
+        output_file = os.path.join(output_path, urlsafe_b64encode(urls[i]))
+        output_file +=  '.' + str(trial_number) + '.pc'
+        try:
+            with open(output_file, 'w') as f:
+                json.dump(output, f)
+        except IOError:
+            raise IOError('Unable to write to {0}'.format(output_file))
+
+def reset_old_files():
+    """Deletes old files to make way for a new run
+
+    Removes:
+    CHROMIUM_SRC/tools/perf/page_sets/url*
+    CHROMIUM_SRC/tools/perf/page_sets/data/url*
+    CHROMIUM_SRC/tools/perf/benchmarks/telemetryBenchmarks.py
+    PLT_SRC/telemetry/data/wpr_source/*
+    PLT_SRC/telemetry/data/results.db
+    PLT_SRC/data/har/*
+    PLT_SRC/telemetry/temp/*
+    """
+    commands = [
+        'rm -f {0}/tools/perf/page_sets/url*'.format(CHROMIUM_SRC),
+        'rm -f {0}/tools/perf/page_sets/data/url*'.format(CHROMIUM_SRC),
+        'rm -f ' \
+        '{0}/tools/perf/benchmarks/telemetryBenchmarks.py'.format(CHROMIUM_SRC),
+        'rm -f data/wpr_source/*',
+        'rm -f temp/*',
+        'rm -f data/results.db',
+        'rm -f {0}/data/har/*'.format(PLT_SRC),
+            ]
+
+    for cmd in commands:
+        p = Popen(cmd, shell=True)
+        p.wait()
+
+def trial_reset():
+    """Deletes files between trials
+
+    Removes:
+    temp/*
+    """
+    commands = [
+        'rm -f temp/*',
+        ]
+
+    for cmd in commands:
+        p = Popen(cmd, shell=True)
+        p.wait()
+
+def aggregate_min_results():
+    """Writes data/results.db with the min of each url and pc url
+
+    Format of output:
+    {'http://jamnoise.com':
+        {'cold_time': 314.159,
+         'modified_cold_time': '300.201'},
+         ...
+    }
+    """
+    aggregated_data = {}
+    data_path = 'temp/*'
+    data_files = [f for f in glob(data_path)]
+    sorted_files = sorted(data_files)
+
+    curr_url = None
+    for tmp_file in sorted_files:
+        data = {}
+        with open(tmp_file) as f:
+            data = json.load(f)
+        if data == {}:
+            raise IOError('Problem reading {0}'.format(tmp_file))
+        trial = int(tmp_file.split('.')[1])
+        url = data.keys()[0]
+        values = data[url]['cold_times']['trial{0}'.format(trial)]
```

Comment:
TODO: On pc data, values is a dict: {'values': [data]} instead of just [data]. I'll look into this soon.